serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/mocking-opnapi-services-with-wiremock.adoc

```diff
@@ -1 +1,310 @@
-//= Mocking OpenAPI Services with Wiremock
\ No newline at end of file
+Mocking OpenAPI services with WireMock
+======================================
+// Metadata:
+:description: Mocking OpenAPI services with WireMock
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, test, wiremock, openapi
+// links:
+:wiremock_url: https://wiremock.org/docs/
+:kogito_examples_url: https://github.com/kiegroup/kogito-examples.git
+:quarkus_test_resource_url: https://quarkus.io/guides/getting-started-testing#quarkus-test-resource
+// Referenced documentation pages.
+:getting-familiar-with-our-tooling: xref:getting-started/getting-familiar-with-our-tooling.adoc
+:create-your-first-workflow-service: xref:getting-started/create-your-first-workflow-service.adoc
+:orchestration-of-openapi-based-services: xref:service-orchestration/orchestration-of-openapi-based-services.adoc
+
+This document describes how to mock OpenAPI services using WireMock. The testing procedure in this document is based on the `serverless-workflow-examples/serverless-workflow-service-calls-quarkus` example application. You can access this example application in link:{kogito_examples_url}[Kogito Examples] GitHub repository.
+
+== Prerequisites
+
+The examples in this document assumes that you have the following prerequisites:
+
+* You have installed the Kogito tooling.
+
++
+--
+For more information about the tooling, see {getting-familiar-with-our-tooling}[Using Serverless Workflow tooling].
+--
+
+* You have created a Serverless Workflow project that orchestrates OpenAPI services.
+
++
+--
+For more information about creating a Serverless Workflow project, see {create-your-first-workflow-service}[Creating your first Serverless Workflow service].
+--
+
++
+--
+For more information about orchestrating OpenAPI services, see {orchestration-of-openapi-based-services}[Orchestrating the OpenAPI services].
+--
+
+== About WireMock
+
+WireMock is an open source API mocking for unit, integration and performance testing. You can use it to isolate your tests from 3rd party APIs and prototype APIs that still don't exist.
+
+For more information about WireMock, see the {wiremock_url}[WireMock documentation].
+
+== Adding a mocked OpenAPI service to your tests
+
+The following procedure describes how to add WireMock to your Serverless Workflow application:
+
+.Procedure
+
+. Add the WireMock dependency to your `pom.xml` file.
+
++
+--
+.WireMock dependency
+[source,xml]
+----
+<dependency>
+    <groupId>com.github.tomakehurst</groupId>
+    <artifactId>wiremock-jre8</artifactId>
+    <version>{wiremock.version}</version>
+    <scope>test</scope>
+</dependency>
+----
+
+[NOTE]
+====
+Replace `{wiremock.version}` with the version of WireMock you want to use.
```

Comment:
is there any version recommended to use with Kogito and Quarkus?

Reply:
Not really. Their API is pretty much stable: https://mvnrepository.com/artifact/com.github.tomakehurst/wiremock

2 years, no updates.